packages/runtime/datastore/src/dataStoreRuntime.ts

```diff
@@ -254,11 +274,17 @@ export class FluidDataStoreRuntime
 		);
 
 		// Validate that the Runtime is compatible with this DataStore.
-		const maybeRuntimeCompatDetails = dataStoreContext as FluidObject<ILayerCompatDetails>;
-		validateRuntimeCompatibility(
-			maybeRuntimeCompatDetails.ILayerCompatDetails,
-			this.dispose.bind(this),
-		);
+		const { ILayerCompatDetails } = dataStoreContext as FluidObject<ILayerCompatDetails>;
+		validateRuntimeCompatibility(ILayerCompatDetails, this.dispose.bind(this));
+
+		if (contextSupportsFeature(dataStoreContext, "setReadonly")) {
+			this._readonly = dataStoreContext.readonly;
```

Comment:
Maybe I'm missing it, but this doesn't seem sufficient to me to match the functionality provided by the DM proxy case.

When DSContext readonly state changes, I don't see where `setReadonlyState` is called.  Maybe I missed it.  Even if you update `get readonly()` to return `dataStoreContext.readonly` if supported (rather than returning the current value of `_readonly`), we won't emit the "readonly" event.

When I saw `setReadonlyState` below, I started looking for a layer compat feature on the DataStore side saying the API exists, so DSContext could call it, but I don't see it. I think that would fix it.

Reply:
Ok yeah I just misremembered the other side and got confused when I tried to double-check.  Looks good.